Reduce clipping through tool and add fade

khooroko opened the pull request:
**Description**
Previously, when swinging too fast, the ball flies off in the opposite direction. This is due to the ball being more than halfway through the tool in one physics timestep (from not being in contact the frame before). The exertion of force in one direction hence causes the ball to bounce off the non-trigger collider component on the tool.

In [5fa7e8e](https://github.com/VRFYP2019/Not-Dodgeball/commit/5fa7e8e7098cb9660ae40a50de3e05de0a080b04), the `isTrigger` field will be turned on for fast movements to prevent this issue.
![smackfast](https://user-images.githubusercontent.com/28438978/64344561-47496000-d021-11e9-96a7-fc3e3475247a.gif)


The work on this inspired [4278799](https://github.com/VRFYP2019/Not-Dodgeball/commit/4278799435eb521858bed4ce09c9e605bfdc2dad), which makes the tool fade on AFK.
![fade](https://user-images.githubusercontent.com/28438978/64344658-7b248580-d021-11e9-803d-3baf632c5e0c.gif)
Note that this requires the rendering mode on the materials involved to not be "opaque".
![image](https://user-images.githubusercontent.com/28438978/64345177-6dbbcb00-d022-11e9-8131-c6f413d326b6.png)

@lyhvictoria

**Impact**
- [x] Medium

**Risks**
- Very mildly inaccurate physics
- May cause game to be more complicated than required (although it adds further incentive to move)

**Test Plan**
Checkout into this branch and hit some balls. Then AFK. Then hit some balls again and repeat.

khooroko committed:
Fix clipping through tool at high speeds

By turning isTrigger on, so that the ball does not bounce off the other
side of the tool

khooroko committed:
Add fade on AFK

lyhvictoria commented:
Is it possible to increase the fade in speed? The fade out speed is ok only the fade in speed. Otherwise, looks good.

khooroko committed:
Address comments

- Reduce unfade time
- Change force exertion to be based on acceleration, scale down amount
of force applied

khooroko committed:
Somewhat account for angular velocity

khooroko commented:
@lyhvictoria 💇‍♂
